### PR TITLE
9. GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by procky66**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -252,6 +252,54 @@ describe("/api/articles/:article_id", () => {
...
+
+		test("status:400, bad request error when passed an invalid article_id", () => {
+			const ARTICLE_ID = "BadKey";
+			return request(app)
+				.get(`/api/articles/${ARTICLE_ID}/comments`)
+				.expect(400)
+				.then(({ body }) => {
+					expect(body.msg).toBe("bad request");
+				});
+		});
```
Comment (xViolaine) [inline __tests__/app.test.js:300]: Another 404 test - article not found

File: db/models/articles.js [hunk 2]
```diff
@@ -46,6 +46,28 @@ exports.fetchArticleById = article_id => {
...
 };
 
+exports.fetchCommentsByArticleId = article_id => {
+	return db
+		.query(
+			`SELECT * FROM comments 
+			WHERE article_id = $1;`,
+			[article_id]
+		)
+		.then(async results => {
```
Comment (xViolaine) [inline db/models/articles.js]: Best practices is not to mix .then and async/await, please refactor this

**Review by xViolaine — CHANGES_REQUESTED** [review]

File: db/models/articles.js [hunk 3]
```diff
@@ -46,26 +46,24 @@ exports.fetchArticleById = article_id => {
 		});
 };
 
-exports.fetchCommentsByArticleId = article_id => {
-	return db
+exports.fetchCommentsByArticleId = async article_id => {
+	results = await db
```
Comment (popatre) [inline db/models/articles.js:50]: `const`

File: db/models/articles.js [hunk 4]
```diff
@@ -46,26 +46,24 @@ exports.fetchArticleById = article_id => {
...
-					`SELECT * FROM articles WHERE article_id = $1;`,
-					[article_id]
-				);
-				if (check.rows.length === 0) {
-					return Promise.reject({ status: 404, msg: "article not found" });
-				}
-			}
-			return results.rows;
-		})
 		.catch(err => Promise.reject(err));
```
Comment (popatre) [inline db/models/articles.js:56]: We ideally only want one catch in the controller.

**Review by popatre — APPROVED** [review]

**Pull request merged**